miner/remote_proposals.go

```diff
@@ -118,33 +119,26 @@ func (pb *RemoteProposalBuilder) Run(ctx context.Context) error {
 			if current <= types.GetEffectiveGenesis() {
 				continue
 			}
-			if err := pb.build(ctx, current); err != nil {
+			if current.FirstInEpoch() {
+				eligibilities = make(map[types.NodeID]map[types.LayerID][]types.VotingEligibility)
+			}
```

Comment:
Now that I look at it, I see a potential rare problem. It might miss this branch, for example when the PC was sleeping on the epoch boundary. It could keep the epoch for which `eligibilities` were calculated and reset if `current.Epoch()` is different.